boot/cmdline.go

```diff
@@ -178,5 +185,31 @@ func ComposeCommandLine(model *asserts.Model) (string, error) {
 		}
 		return "", err
 	}
-	return mbl.CommandLine(nil)
+	bv, err := mbl.GetBootVars("snapd_extra_cmdline_args")
+	if err != nil {
+		return "", fmt.Errorf("cannot load command line arguments from boot environment: %v", err)
+	}
+	modeArgs := "snapd_recovery_mode=run"
+	return mbl.CommandLine(bv["snapd_extra_cmdline_args"], modeArgs)
+}
+
+func SetExtraCommandLineArgs(model *asserts.Model, args string) error {
+	if model.Grade() == asserts.ModelGradeUnset {
+		return nil
+	}
+	// get the ubuntu-boot bootloader
+	opts := &bootloader.Options{
+		NoSlashBoot: true,
+	}
```

Comment:
should we maybe instead pass in a *bootloader.Options to this function? It's not clear to me that we will always want to use this function on ubuntu-boot, and it's also not clear to me that we will always have `NoSlashBoot: true` when we are referring to the ubuntu-boot bootloader.

Reply:
I don't think we're passing bootloader.Options as input to any of the functions in boot package. OTOH, assuming this is called in run mode, we can probably do the same thing as in #8947 and use `bootloader.Find("", nil)`

Reply:
passing comment, I haven't looked at this yet, will this not also be called from install mode though?

Reply:
it's a bit hard to judge the correctness of this without understanding where it's going to be called from

Reply:
the name of this feels a bit lightweight considering that this will need a reseal done before being invoked. OTOH the final API because of reseal might be very different anyway.

Reply:
It's my understanding that this should also be called from install mode to do the initial sealing.